### PR TITLE
Adds `ServerDriveQueryInformationRequest`

### PR DESCRIPTION
**Pull request by ibeckermayer**
Also changes `FilesystemRequest` to be named `ServerDriveIoRequest`.

### CODE REVIEW

File: crates/ironrdp-rdpdr/src/pdu/efs.rs [hunk 1]
```diff
@@ -1589,3 +1597,250 @@ bitflags! {
         const FILE_OVERWRITTEN = 0x00000003;
     }
 }
+
+/// [2.2.3.3.8 Server Drive Query Information Request (DR_DRIVE_QUERY_INFORMATION_REQ)]
+///
+/// Note that Length, Padding, and QueryBuffer fields are all ignored in keeping with the [analogous code in FreeRDP].
+///
+/// [2.2.3.3.8 Server Drive Query Information Request (DR_DRIVE_QUERY_INFORMATION_REQ)]: https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-rdpefs/e43dcd68-2980-40a9-9238-344b6cf94946
```
Comment (CBenoit) [inline crates/ironrdp-rdpdr/src/pdu/efs.rs:1605]: Nit: you can ignore if you want, but I was thinking about only putting a link on the section number instead:
```suggestion
/// [2.2.3.3.8] Server Drive Query Information Request (DR_DRIVE_QUERY_INFORMATION_REQ)
///
/// Note that Length, Padding, and QueryBuffer fields are all ignored in keeping with the [analogous code in FreeRDP].
///
/// [2.2.3.3.8]: https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-rdpefs/e43dcd68-2980-40a9-9238-344b6cf94946
```
This requires less duplication and results in shorter lines. What do you think?

File: crates/ironrdp-rdpdr/src/pdu/efs.rs [hunk 2]
```diff
@@ -1589,3 +1597,250 @@ bitflags! {
...
+/// [2.4.7 FileBasicInformation [MS-FSCC]]: https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-fscc/16023025-8a78-492f-8b96-c873b042ac50
+#[derive(Debug, Clone)]
+pub struct FileBasicInformation {
+    pub creation_time: i64,
+    pub last_access_time: i64,
+    pub last_write_time: i64,
+    pub change_time: i64,
+    pub file_attributes: FileAttributes,
+    // NOTE: The `reserved` field in the spec MUST not be serialized and sent over RDP, or it will break the server implementation.
+    // FreeRDP does the same: https://github.com/FreeRDP/FreeRDP/blob/1adb263813ca2e76a893ef729a04db8f94b5d757/channels/drive/client/drive_file.c#L508
```
Comment (CBenoit) [inline crates/ironrdp-rdpdr/src/pdu/efs.rs:1747]: Thank you for documenting this :pray:

**Review by CBenoit — APPROVED** [review]
LGTM. Leaving some time if you want to address the comment about links, but otherwise good to merge.

File: crates/ironrdp-rdpdr/src/pdu/efs.rs (hunk 1 above)
Comment (ibeckermayer) [reply crates/ironrdp-rdpdr/src/pdu/efs.rs:1605]: I'm fine with that style in general, but I'd be bothered by the inconsistency with the rest of [this part of] the codebase if I were to fix just this one. Seems more trouble than it's worth to fix them all at this point.

Comment (CBenoit) [reply crates/ironrdp-rdpdr/src/pdu/efs.rs:1605]: Fear not! [`sd`](https://github.com/chmln/sd) is here to save the day!
I’ll run that in another PR though, you’re right.

Comment (CBenoit) [reply crates/ironrdp-rdpdr/src/pdu/efs.rs:1605]: https://github.com/Devolutions/IronRDP/pull/247

**Pull request merged**